docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

```diff
@@ -0,0 +1,250 @@
+      plan: hobbyist-2
+      type: pg
+      maintenance:
+        day: 2
+        hour: 22
+        minute: 0
+      backup:
+        hour: 20
+        minute: 0
+    vshn:
+      minorVersion: '4'
+----
+
+Advantages::
+
+* We can use the same XRD
+* Adding new providers to the list is easy
+
+Disadvantages::
+
+* We don't really have a "common look"
+* The customer needs to know what the underlying service provider is, even for simple instances
+
+=== Proposal 2: Separate XRDs
+
+This approach will have different XRDs for each provider.
+However, they will follow the same structure, as far as possible.
+All the toplevel fields will be the same, the differences will be apparent in the sub-fields.
+They will also have the same kind, but a different APIVersion.
+
+.Example Claim
+[source, yaml]
+----
+# Exoscale example
+apiVersion: exoscale.appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    service:
+      version: '14'
+      type: pg
+      settings: {}
+    size:
+      plan: hobbyist-2
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+
+# Converged example
+apiVersion: vshn.appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    service:
+      version: '14.4'
+      settings: {}
+      pgbouncer:
+        settings: {}
+    size:
+      cpu: 2
+      memory: 8Gi
+      disk: 200Gi
+      encrypted: true
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+----
+
+Advantages::
+
+* Same look between all providers, deploying PostgreSQL to any backend feels familiar
+* Specifics can be handled in a sensible way
+
+Disadvantages::
+
+* The user could get confused by objects with the same kind
+
+=== Proposal 3: Composed XRDs
+
+There's a base XRD that handles the common configurations.
+All provider specific parameters are put in a separate XRD.
+
+This idea may not be possible with Crossplane, but it's here for completeness’ sake.
+
+.Example Claim
+[source, yaml]
+----
+apiVersion: appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    version: '14'
+    pgSettings: {}
+
+
+apiVersion: exoscale.appcat.vshn.io/v1
+kind: PostgreSQLConfig
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    instanceRef: my-instance
+    plan: hobbyist-2
+    type: pg
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+----
+
+Advantages::
+
+* Clear separation of base and specific configuration parameters
+
+Disadvantages::
+
+* "same look" will only apply to the base XRDs
+* Pretty complicated to use for the end-user
+
+=== Proposal 4: Same XRD and Parse the Input
+
+We use the same XRD for every provider.
+Most fields are simply strings, and they will then be parsed in the composition.
```

Comment:
I don't understand this proposal. How is it different from Proposal 1?

Reply:
The difference is, that in proposal 1 there would be some provider specific fields. Here, we don't have any specific fields at all.

Reply:
Does that mean the user has to provide the correct string, resp. yaml/json?

Reply:
Yes, exactly.

I've added the source for this idea to the text as well.